crates/services/compression/src/service.rs

```diff
@@ -362,7 +390,7 @@ where
 
     async fn shutdown(mut self) -> anyhow::Result<()> {
         // gracefully handle all the remaining blocks in the stream and then stop
-        while let Some(block_with_metadata) = self.block_stream.next().await {
+        if let Some(Some(block_with_metadata)) = self.block_stream.next().now_or_never() {
```

Comment:
If we're in the process of shutting down, why not rely that the stream will also get the shutdown signal at some point and close on its own?

Is potentially swallowing up an update or two here ok when shutting down?